pkg/api/types.go

```diff
@@ -3577,8 +3580,78 @@ type Event struct {
 	// Type of this event (Normal, Warning), new types could be added in the future.
 	// +optional
 	Type string
+
+	// Time when this Event was first observed.
+	// +optional
+	EventTime metav1.MicroTime
+
+	// Data about the Event series this event represents or nil if it's a singleton Event.
+	// +optional
+	Series *EventSeries
+
+	// Component that generated the Event.
+	// +optional
+	Origin EventSource
+
+	// What Origin did/failed to do.
+	// +optional
+	Action EventAction
+
+	// On what Origin acted upon.
+	// +optional
+	Object *ObjectReference
+
+	// Optional secondary Object for more complex actions.
+	// +optional
+	SecondaryObject *ObjectReference
+
+	// +optional
+	Severity EventSeverity
```

Comment:
Here again maybe we can just reuse `Type` above which seems to be similar?
Is the reason for defining these extra fields just so we keep populating the old ones as they are? Because it seems to me like we're going to populate these fields with the same values too (at least `Object` and `Origin` I guess). I might be missing something here.